bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,291 @@
+We are defining here a workflow that leverages the hollow jar packaging and server deployment scanner capabilities.
+
+===== Dev mode
+
+* A dev server is an hollow server scanning the directory target/deployments.
+* A dev app is the primary artifact copied to the target/deployments dir.
+
+The plugin attempts to force the provisioning of the deployment scanner in dev mode:
+
+* If deployment-scanner is excluded, remove it from exclusion (Warn).
+* Add the deployment-scanner layer to the set of layers (Info)
+* If a provisioning.xml file is set (and no layers override it), a warning is displayed advertising that we can’t enforce the presence of the scanner. (Warn)
+* If the deployment-scanner is not in the config, the CLI operation will abort the plugin execution. (Error)
+
+===== Workflow examples
+
+Description of the 2 workflows (with and without dev mode). These examples could be designed differently according to the content of the pom file.
+
+Nominal mode, full repackaging of the server done for each rebuild, server restarted after each rebuild (not viable):
+
+* mvn package ⇒ full repackaging
+* mvn wildfly-jar:run
+* (kill synchronous execution).
+* User make changes
+* mvn package ⇒ full repackaging
+* ...
+
+Dev mode, server built/started once, app automatically re-deployed.
+
+* mvn wildfly-jar:dev ⇒ hollow server built and started
+* mvn package -Ddev ⇒ Fast, no packaging, app copied to deployments dir. Application automatically deployed.
+* User make changes
+* mvn package -Ddev ⇒ Fast, no packaging, app copied to deployments dir. Application automatically re-deployed.
+* ...
+* mvn wildfly-jar:shutdown
+* User is fine with his changes, he can do the server+app packaging: mvn package
+
+NB: This workflow doesn’t require support in IDE, it is 100% maven. It could be optimized with IDE plugin (eg: netbeans plugin to track static 
+files and avoid to re-package in this case).
+
+
+==== Cloud context
+
+The plugin allows to generate a bootable jar usable in various Cloud execution contexts: custom container, Google JIB, Java s2i build.
+
+=== Nice-to-Have Requirements
+
+* Add the ability to copy content in the server during build. That is a common requirement (eg: auth properties files).
+* Ability to generate a runtime maven repository in order to resolve the server artifacts from maven local cache. 
+Huge benefit in term of jar size and boot time (4/5 time faster, around 160ms to start vs 700 ms). 
+Specifically in a docker/openshift context, relying on slim server + maven repository speeds up startup without impacting image size.
+Eg: java -Dmaven.repo.local=/maven-repo -jar myapp-wildfly.jar
+
+
+=== Non-Requirements
+
+* Offer a new way to configure the server (eg: Thorntail yaml file).
+* Usage of a different log manager than JBoss log manager is out of scope.
+* No domain support.
+
+== Implementation Plan
+
+* Evolve wildfly-core with a runtime to boot the bootable jar.
+* Develop new maven plugin.
+
+== Test Plan
```

Comment:
Could you please add links to the affected test suites/repos?

Reply:
Done